batchflow/models/validator/validator.py

```diff
@@ -0,0 +1,337 @@
+                ...
+        pretrained: (optional)
+            path: <model_path>
+        test: (optional)
+            dataset:
+                - <dataset_param_0>: <value_0>
+                ...
+            metrics: (metics from batchflow)
+                - <classification|segmentation|mask|instance|regression>
+                    <classification_kwarg_0>: <value_0>
+                    <classification_kwarg_1>: <value_1>
+                    ...
+                    evaluate:
+                        - <metric_name_0>:
+                            <metric_kwarg_0>: <value_0>
+                            <metric_kwarg_1>: <value_1>
+                        - <metric_name_1>:
+                            ...
+                        ...
+            custom_metrics: (optional, metrics defined in ModelAPI child-class)
+                - <custom_metric_0>
+                - <custom_metric_1>
+                ...
+        ```
+    name : str
+        name of the current model (from yaml)
+    task : str
+        name of the current task (from yaml)
+    """
+    def __init__(self, config=None, **kwargs):
+        if isinstance(config, str):
+            self.config_path = config
+            with open(config) as file:
+                self.config = yaml.load(file, Loader=yaml.Loader)
+        elif isinstance(config, dict):
+            self.config_path = None
+            self.config = deepcopy(config)
+        elif isinstance(config, Config):
+            self.config_path = None
+            self.config = deepcopy(config.config)
+        elif config is None:
+            self.config = {}
+
+        self.config = {**self.config, **kwargs}
+
+        if 'train' in self.config and 'pretrained' in self.config:
+            raise ValueError("Both 'train' and 'pretrained' was founded.")
+
+        self._pretrained = self.config.get('pretrained', {})
+        if 'pretrained' in self.config:
+            if self._pretrained is None:
+                self._pretrained = {}
+            elif isinstance(self._pretrained, str):
+                self._pretrained = {'path': self._pretrained}
+
+        self.config['train'] = self.config.get('train', {})
+        self.config['test'] = self.config.get('test', {})
+
+        self._train_ds = self.config.get('train_dataset', {})
+        if isinstance(self._train_ds, str):
+            self._train_ds = {'path': self._train_ds}
+        self._test_ds = self.config.get('test_dataset', {})
+        if isinstance(self._test_ds, str):
+            self._test_ds = {'path': self._test_ds}
+
+        self._metrics = self.config.get('metrics', [])
+        if isinstance(self._metrics, str):
+            self._metrics = [item.strip() for item in self._metrics.split(',')]
+
+        self._cv = self.config.get('cv', {})
+        if self._cv is None:
+            self._cv = {}
+        elif isinstance(self._cv, str):
+            self._cv = {'path': self._cv}
+
+        self.train_dataset = None
```

Comment:
`self.train_dataset` and `self.test_dataset`  are not used anywhere. I suppose they are here to demonstrate the possibility to use such attributes as transport for entities; though, I would highly discourage it as it is very implicit.

Reply:
I've removed them